File: 03.programmers/2019_kakao_winter_internship/02.Python/02.py
def solution(s):
    answer = []
    s = s[2:-2]
    s = s.split("},{")
    s.sort(key = len)
    for i in s:
        ii = i.split(',')
        for j in ii:
            if int(j) not in answer:
                answer.append(int(j))
    return answer

input_s = "{{2},{2,1},{2,1,3},{2,1,3,4}}" 
#input_s = "{{4,2,3},{3},{2,3,4,1},{2,3}}"
#input_s = "{{123}}"
#input_s = "{{20,111},{111}}"
print(solution(input_s))

File: 2023/Programmers/lv01/javascript/공원산책/02.js
function solution(park, routes) {
  var answer = [];
  const n = park.length;
  const m = park[0].length;

  const dy = { N: -1, E: 0, S: 1, W: 0 };
  const dx = { N: 0, E: 1, S: 0, W: -1 };

  let y = park.findIndex((y) => y.includes("S"));
  let x = park[y].split("").findIndex((x) => x === "S");

  for (let route of routes) {
    const [dir, len] = route.split(" ");
    let nlen = Number(len);

    let cy = y;
    let cx = x;
    let ny;
    let nx;
    while (nlen--) {
      (ny = y + dy[dir]), (nx = x + dx[dir]);

      if (ny >= 0 && ny < n && nx >= 0 && nx < m && park[ny][nx] !== "X") {
        (y = ny), (x = nx);
      } else {
        (y = cy), (x = cx);
        break;
      }
    }
  }
  answer = [y, x];

  return answer;
}
